src/server/protocol_client.cc

```diff
@@ -212,7 +219,7 @@ ProtocolClient::~ProtocolClient() {
 #endif
 }
 
-error_code ProtocolClient::ResolveMasterDns() {
+error_code ProtocolClient::InitEndpointWithDns() {
```

Comment:
The fact that you set the server_context_.endpoint is internal implementation of this class. I dont think the function name should expose this. I think ResolveDns is better

Reply:
Actually a better api to this class can be Just connect function.
The internal will be if this funcion will be if this class was initialized from host name it will call resolve dns. If it was created from ip skip resolve dns